src/schemas/user.js

```diff
@@ -12,6 +16,38 @@ const userSchema = new mongoose.Schema({
     required: true,
     unique: true
   },
+  username: {
+    type: String,
+    required: true
+  },
+  skills: {
+    type: [Schema.Types.ObjectId], // department
+    default: [],
+    required: true
+  },
+  knowledge: {
+    type: [Schema.Types.ObjectId], // department
+    default: [],
+    required: true
+  },
+  mentor: {
+    type: Schema.Types.ObjectId,
+    required: false
+  },
+  mentorshipCertified: {
+    type: Boolean, // department
+    required: false,
+    default: false
+  },
+  emailVerified: {
+    type: Boolean, // department
+    required: false,
+    default: false
+  },
+  verificationCode: {
+    type: String, // department
+    required: true
```

Comment:
Should this be really `required`? You cannot create a user without a this?

Reply:
not needed